ngraph/core/reference/CMakeLists.txt

```diff
@@ -21,6 +21,20 @@ file(GLOB_RECURSE PUBLIC_HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/include/*.hpp)
 
 set(REF_IMPL_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/include")
 
+include(FetchContent)
+
+FetchContent_Declare(
+    xbyak
+    GIT_REPOSITORY https://github.com/herumi/xbyak.git
```

Comment:
Why do you use thirdparty repo for xbyak instead of using one from mkldnn?

Reply:
There are several reasons for this.
First, xback is transitively added to the CPU plugin via the MKLDNN library. The MKLDNN library is privately linked. So xbyak is internal dependency of MKLDNN library and I can't make it public.
Secondly, nGraph and transformations libraries shouldn't dependent from plugins internal libraries.
It seems that it is possible to extract xbyak into a common libraries space but I think it isn't good idea because xbyak inside MKLDNN library is modified and customized for MKLDNN library.

Reply:
IE core already depends on xbyak from mkl-dnn: https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/inference_engine/CMakeLists.txt#L123
I don't see any big issues with this approach, as xbyak is header-only library and doesn't introduce any binary dependencies. 

Anyway, I believe IE core and ngraph should use the same xbyak sources.

@dmitry-gorokhov  @ilya-lavrenov what do you think?

Reply:
>IE core already depends on xbyak from mkl-dnn:
This looks like a workaround for the object library.

Reply:
> Anyway, I believe IE core and ngraph should use the same xbyak sources.

I don't mind on both approaches. 

> This looks like a workaround for the object library.

No, we just need xbyak in IE to understand CPU traits.

Reply:
Currently, we have 2 gtests as well (ngraph and IE folders). Maybe we can introduce thirdparty folder on root level and IE and ngraph will reuse it? In this case xbyak, gtests are good candidates. How do you think?

CC @ilyachur

Reply:
I think it's a good idea. It allows us to avoid using xbayk directly from the internal directory of mkldnn library for IE.

Reply:
done

Reply:
thanks!

Reply:
welcome (: